src/phase_1_image_preprocessing/scratch_directory_manager.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Nov 28 11:55:15 2018
+
+Scratch_Directory_Manager.py -add fname -remove fname -lookup fname --remove_all --verbose
+
+@author: mmacferrin
+"""
+
+#######################################################################################
+## Code for importing the parent directory in order to get the File_Locations object
+import os,sys,inspect
+currentdir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+parentdir = os.path.dirname(currentdir)
+sys.path.insert(0,parentdir)
+## Also add the utility folder for the csv names
+utilitydir = os.path.join(parentdir, "utilities")
+sys.path.insert(0,utilitydir)
+#######################################################################################
+
+from file_locations import ASIFT_Scratch_Dir, ASIFT_Scratch_Logfile
+from csv_auto_reader import read_simple_csv
+from csv_writer import write_csv
+
+import hashlib
+import os
+import shutil
+import numpy
+import argparse
+
+class SCRATCH_Manager:
+
+    def __init__(self, scratch_dir=ASIFT_Scratch_Dir, scratch_logfile=ASIFT_Scratch_Logfile):
+        self.scratch_dir = scratch_dir
+        self.logfile = scratch_logfile
+        self.logfile_data = None
+        self.logfile_numpy_dt = None
+        return
+
+    def hash_filename(self, fname, remove_path=True):
+        '''Take a filename, generate a hash string. Right now using the md5 algorithm.
+        For this purpose we are not worried about security vulnerabilities of md5,
+        it works to create unique strings for our purposes.
+
+        If 'remove_path' set, it will get only the file name, not the entire path to the file if it's included.
+
+        Returns a 32-character hexadecimal string hash value, with the first four letters of filename and an underscore pre-pended.'''
+        if remove_path:
+            fname = os.path.split(fname)[-1]
+
+        m = hashlib.md5()
+        m.update(fname)
+        hashstr = m.hexdigest()
+
+        fname_base = os.path.splitext(fname)[0]
+        hash_name = (fname_base[0:4] if len(fname_base) >= 4 else fname_base) + '_' + hashstr
+        return hash_name
+
+    def read_logfile(self):
+        '''Read the logfile and return a numpy data array from 'read_simple_csv'''
+        # We artificially make the string length longer in the numpy data type,
+        # so we don't accidentally concatenate long filenames added to the database.
+        # Max filename size here, 2048 characters.
+        self.logfile_data = read_simple_csv(self.logfile, string_length_size=2048)
+        self.logfile_numpy_dt = self.logfile_data.dtype
```

Comment:
getting `AttributeError: 'NoneType' object has no attribute 'dtype' `  in this line